### PR TITLE
[SYCL][NATIVECPU] Support atomic fence on Native CPU

### PR DESCRIPTION
**Pull request by PietroGhg**
Adds `__spirv_MemoryBarrier` to the Native CPU device library.
UR PR: https://github.com/oneapi-src/unified-runtime/pull/1874

### CODE REVIEW

File: libdevice/nativecpu_utils.cpp
Comment (steffenlarsen) [inline libdevice/nativecpu_utils.cpp:1]: Though I somewhat agree that 100644 is more correct than 100755, it seems unrelated to these changes, so please revert the permissions.

File: sycl/test/check_device_code/native_cpu/native_cpu_attrs.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#include "sycl.hpp"
+using namespace sycl;
+
+class Test;
+int main() {
+  sycl::queue deviceQueue;
+  sycl::nd_range<2> r({1, 1}, {
+                                   1,
+                                   1,
+                               });
```
Comment (steffenlarsen) [inline sycl/test/check_device_code/native_cpu/native_cpu_attrs.cpp:13]: This is some odd formatting. I think the formatter is broken, so could you please make sure the formatting is in line with clang-format?

**Review by steffenlarsen — APPROVED** [review]
Small comments and otherwise it LGTM.

File: libdevice/nativecpu_utils.cpp
Comment (PietroGhg) [reply libdevice/nativecpu_utils.cpp:1]: Thanks, I'll create a separate PR to fix the file permissions, I'm not sure why the file is marked as executable but it's not required to be.

**Review by steffenlarsen — APPROVED** [review]

**Review by uwedolinsky — APPROVED** [review]

**Comment by omarahmed1111** [comment]
@PietroGhg UR PR merged, please update UR tag. Thanks!

**Review by omarahmed1111 — APPROVED** [review]
